experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -202,7 +203,14 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 
 			// If this transaction is not nested, add it to the undo commit tree
 			if (!this.isTransacting()) {
-				this.undoRedoManager?.trackCommit(this.head, UndoRedoManagerCommitType.Undoable);
+				const repairData = this.undoRedoManager?.trackCommit(
+					this.head,
+					UndoRedoManagerCommitType.Undoable,
+				);
+
+				if (repairData !== undefined) {
+					this.repairStore.set(this.head.revision, repairData);
+				}
```

Comment:
Nit: since we're forced to have some explicit if check, it would be a little less confusing to check the `undoRedoManager`:
```suggestion
				if (this.undoRedoManager !== undefined) {
					const repairData = this.undoRedoManager.trackCommit(
						this.head,
						UndoRedoManagerCommitType.Undoable,
					);
					this.repairStore.set(this.head.revision, repairData);
				}
```

Reply:
Nice, did this in latest commit.